swap/src/main.rs

```diff
@@ -14,25 +14,31 @@
 #![allow(non_snake_case)]
 
 use anyhow::{bail, Context, Result};
+use cli::{Command, Options, Resume};
+use config::Config;
+use database::{Database, Swap};
 use libp2p::{core::Multiaddr, PeerId};
+use network::transport::build;
 use prettytable::{row, Table};
+use protocol::{alice, alice::AliceState, bob, bob::BobState, SwapAmounts};
 use rand::rngs::OsRng;
 use std::sync::Arc;
 use structopt::StructOpt;
-use swap::{
-    bitcoin,
-    cli::{Command, Options, Resume},
-    config::Config,
-    database::{Database, Swap},
-    monero,
-    network::transport::build,
-    protocol::{alice, alice::AliceState, bob, bob::BobState},
-    trace::init_tracing,
-    SwapAmounts,
-};
+use trace::init_tracing;
 use tracing::{info, log::LevelFilter};
 use uuid::Uuid;
 
+pub mod bitcoin;
+pub mod cli;
+pub mod config;
+pub mod database;
+pub mod monero;
+pub mod network;
+pub mod protocol;
+#[cfg(test)]
```

Comment:
That's what I am talking about :)

Excellent, now I can go and isolate our `run_until` function that only exists for test purposes :)

Reply:
we are calling `run_until()` in `swap()`. How would you remove `run_until()` from production code without causing code duplication?

Reply:
> we are calling run_until() in swap(). How would you remove run_until() from production code without causing code duplication?

I am happy to take over this change myself at a later stage.